camel/models/samba_model.py

```diff
@@ -0,0 +1,159 @@
+        self._url = url or os.environ.get(
+            "SAMBA_API_BASE_URL",
+            "https://fast-api.snova.ai/v1/chat/completions",
+        )
+        self._token_counter = token_counter
+        self.model_config_dict = model_config_dict
+        self.check_model_config()
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        r"""Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(ModelType.GPT_3_5_TURBO)
+        return self._token_counter
+
+    def check_model_config(self):
+        r"""Check whether the model configuration contains any
+        unexpected arguments to SambaNova API.
+
+        Raises:
+            ValueError: If the model configuration dictionary contains any
+                unexpected arguments to SambaNova API.
+        """
+        for param in self.model_config_dict:
+            if param not in SAMBA_API_PARAMS:
+                raise ValueError(
+                    f"Unexpected argument `{param}` is "
+                    "input into SambaNova model backend."
+                )
+
+    @api_keys_required("SAMBA_API_KEY")
+    def run(  # type: ignore[misc]
+        self, messages: List[OpenAIMessage]
+    ) -> Stream[ChatCompletionChunk]:
+        r"""Runs SambaNova's FastAPI service.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            Stream[ChatCompletionChunk]: A stream of chat completion chunks
+                from the model.
+
+        Raises:
+            ValueError: If the model is not configured to run in streaming
+                mode.
+        """
+        if self.model_config_dict.get("stream") is not True:
+            raise ValueError("SambaNova only supports streaming mode")
+
+        import httpx
+
+        headers = {
+            "Authorization": f"Basic {self._api_key}",
+            "Content-Type": "application/json",
+        }
+
+        data = {
+            "messages": messages,
+            "max_tokens": self.token_limit,
+            "stop": self.model_config_dict.get("stop"),
+            "model": self.model_type.value,
+            "stream": self.model_config_dict.get("stream"),
+            "stream_options": self.model_config_dict.get("stream_options"),
+        }
+
+        with httpx.stream(
+            "POST",
+            self._url or "https://fast-api.snova.ai/v1/chat/completions",
+            headers=headers,
+            json=data,
+        ) as response:
+            stream = Stream[ChatCompletionChunk](
+                cast_to=ChatCompletionChunk, response=response, client=OpenAI()
+            )
+            for chunk in stream:
+                yield chunk
```

Comment:
Can we accumulate all the chunks generated by yield and assemble them into the final response and return it to the user?， just a hypothesis， if user set stream value is false